src/rx_signal_processing.py

```diff
@@ -20,26 +19,25 @@
 from functools import reduce
 
 try:
-    import cupy as cp
-except:
+    import cupy as xp
+except ImportError:
     cupy_available = False
+    import numpy as xp
 else:
     cupy_available = True
 
-if cupy_available:
-    import cupy as xp
-else:
-    import numpy as xp
+# Import some additional modules here. This is avoided on import of this module to facilitate testing of the DSP class.
+if __name__ == '__main__':
+    import posix_ipc as ipc
 
-sys.path.append(os.environ['BOREALISPATH'])
-if __debug__:
-    from build.debug.src.utils.protobuf.rxsamplesmetadata_pb2 import RxSamplesMetadata
-else:
-    from build.release.src.utils.protobuf.rxsamplesmetadata_pb2 import RxSamplesMetadata
+    sys.path.append(os.environ['BOREALISPATH'])
 
-from utils.message_formats import ProcessedSequenceMessage, DebugDataStage, OutputDataset
-from utils.options import Options
-from utils import socket_operations as so
+    if __debug__:
+        from build.debug.src.utils.protobuf.rxsamplesmetadata_pb2 import RxSamplesMetadata
+    else:
+        from build.release.src.utils.protobuf.rxsamplesmetadata_pb2 import RxSamplesMetadata
+
+    from utils.message_formats import ProcessedSequenceMessage, DebugDataStage, OutputDataset
 
 
```

Comment:
All the core modules have the logging imports at the bottom of the script in the solo `if __name__ == '__main__':` block, I think for consistency we should leave this file the same way. I did add another block to the top of this file, so I guess I have already broken that consistency. I think I would prefer changing it back, and only having one block at the bottom of the file. It seems messy to have more than one  `if __name__ == '__main__':` block.

Alternatively, we could break out DSP into its own file again. It used to be, but we merged it into rx_signal_processing since it was the only place DSP was used. That's not the case anymore, so there is an argument that it deserves its own file.